### PR TITLE
Add the plumbing for enabling variable auto generation via env var docs

### PR DESCRIPTION
**Pull request by jessbumblebee**
This PR adds a new script "run" which runs before run.py.
The purpose is to download and install the env-var-docs python package.
We create a python virtual environment(venv), install the dynamically downloaded env-var-docs package in it and run our python deploy code inside the venv so it has access to env-var-docs.

#132 contains the feature this would enable, whose longer term goal is to automatically generate the variables that allow configuring the server via the deploy system.
Note that this PR does not yet enable the feature. It merely provides the script that can do so. 
When we are happy to enable the feature, we should replace all calls to run.py with run, which then calls run.py in turn.

Some reasoning about the location of this: 
We need to enable the python packages across all deployment implement system (dockerized deployment, checkout deployment, AWS. Azure). The run.py script is the first step that is shared between all those options. There are some other benefits to placing it here rather than somewhere in the deploy repository (e.g. you can use run during development to not have to code across multiple branches) 
We can not enable the package in run.py directly, because the activation of the python virtual environment needs to happen in the parent process which the python process is started from before the code is started.

### CODE REVIEW

File: cloud/shared/bin/python_env_setup [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  local requirements_file_path=$1
+
+  # Check if there are any requirements to install
+  if [[ ! -f "$requirements_file_path" ]]; then
+    echo "Requirements file $requirements_file_path does not exist, python setup skipped"
+    return
+  fi
+
+  # Create a virtual environment if it doesn't exist yet
+  if [[ ! -d .venv ]]; then
```
Comment (avrittrohwer) [inline cloud/shared/bin/python_env_setup:25]: The scripts in civiform-deploy remove the checkout directory on every run: https://github.com/civiform/civiform-deploy/blob/main/bin/lib/checkout.sh#L80.  This was due to the checkout directory getting larger on each run: https://github.com/civiform/civiform-deploy/pull/31.

If we put the venv directory in the checkout directory, with the current civiform-deploy scripts, we will re-download the env-var-docs parser package on every deployment tool run.  This adds 5-10 seconds of latency to every run.  What do you think about modifying the civiform-deploy script to not remove the .venv directory?

File: cloud/shared/bin/run [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env bash
+# Script for running the commands supported by the deploy system by calling run.py.
+# See comments below for why we need this scripts rather than running run.py directly.
+# TODO(#4612) Enable the env-var-docs variable generation feature by using this script
+# instead of run.py across the civiform-deploy repository
+
+# In the server code repository in git(civiform), we define a python package called env-var-docs,
```
Comment (rockycodes) [inline cloud/shared/bin/run:7]: suggestion (non-blocking): link to package in comment https://github.com/civiform/civiform/tree/main/env-var-docs

File: cloud/shared/bin/run [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# whose purpose is to automate various aspects of adding a variable to application.conf.
+# Variables in applicaiton.conf make the server configurable.
```
Comment (rockycodes) [inline cloud/shared/bin/run:9]:
```suggestion
# Variables in application.conf make the server configurable.
```

File: cloud/shared/bin/run [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env bash
+# Script for running the commands supported by the deploy system by calling run.py.
+# See comments below for why we need this scripts rather than running run.py directly.
```
Comment (rockycodes) [inline cloud/shared/bin/run:3]:
```suggestion
# See comments below for why we need this script rather than running run.py directly.
```

**Review by rockycodes — COMMENTED** [review]
Looks good to me! I don't feel like I understand it quite well enough to approve, so I leave that to Avritt, but I left a couple of comments for readability.

File: cloud/shared/bin/python_env_setup (hunk 1 above)
Comment (jessbumblebee) [reply cloud/shared/bin/python_env_setup:25]: Good catch, I'll look into it in a separate PR as it needs to be done in the other repository.
We could either exclude the .venv from the removal or place the .venv elsewhere, I'll have a think about it.

File: cloud/shared/bin/run (hunk 4 above)
Comment (jessbumblebee) [reply cloud/shared/bin/run:3]: Done

File: cloud/shared/bin/run (hunk 3 above)
Comment (jessbumblebee) [reply cloud/shared/bin/run:9]: Done.

File: cloud/shared/bin/run (hunk 2 above)
Comment (jessbumblebee) [reply cloud/shared/bin/run:7]: Done.

File: cloud/shared/bin/python_env_setup (hunk 1 above)
Comment (jessbumblebee) [reply cloud/shared/bin/python_env_setup:25]: Ps: as the PR does not yet enable the feature, the latency won't have an impact.

**Review by avrittrohwer — APPROVED** [review]

**Pull request merged**